### PR TITLE
move grab related functionalities from Robot to KinBody

### PR DESCRIPTION
**Pull request by kanbouchou**
Need this because we want to do something like

1. robot grabs a mug
2. robot releases the mug to a table, a KinBody object
3. robot moves the table (and then the mug grabbed by the table also moves with it)

tested from python bindings

- robot can grab/release and GetGrabbed() returns proper information
- KinBody can grab/release and GetGrabbed() returns proper information

### CODE REVIEW

File: python/bindings/openravepy_kinbody.cpp [hunk 1]
```diff
@@ -2985,6 +3070,15 @@ void init_openravepy_kinbody()
                        .def_pickle(JointInfo_pickle_suite())
     ;
 
+    object grabbedinfo = class_<PyKinBody::PyGrabbedInfo, boost::shared_ptr<PyKinBody::PyGrabbedInfo> >("GrabbedInfo", DOXY_CLASS(KinBody::GrabbedInfo))
+                         .def_readwrite("_grabbedname",&PyKinBody::PyGrabbedInfo::_grabbedname)
+                         .def_readwrite("_bodylinkName",&PyKinBody::PyGrabbedInfo::_bodylinkName)
```
Comment (cielavenir) [inline python/bindings/openravepy_kinbody.cpp]: I think this has to be _robotlinkname. This is Python interface.

File: python/databases/linkstatistics.py [hunk 2]
```diff
@@ -198,7 +198,7 @@ def generate(self,**kwargs):
     def _GetJointSpheresFromGrabbed(self, grabbedinfo):
         for testgrabbedinfo, testjointspheres in self.grabbedjointspheres:
             if len(testgrabbedinfo) == len(grabbedinfo):
-                if all([(grabbedinfo[i]._grabbedname == testgrabbedinfo[i]._grabbedname and grabbedinfo[i]._robotlinkname == testgrabbedinfo[i]._robotlinkname and sum(abs(grabbedinfo[i]._trelative-testgrabbedinfo[i]._trelative)) <= 1e-7 and grabbedinfo[i]._setRobotLinksToIgnore == testgrabbedinfo[i]._setRobotLinksToIgnore) for i in range(len(grabbedinfo))]):
+                if all([(grabbedinfo[i]._grabbedname == testgrabbedinfo[i]._grabbedname and grabbedinfo[i]._bodylinkName == testgrabbedinfo[i]._bodylinkName and sum(abs(grabbedinfo[i]._trelative-testgrabbedinfo[i]._trelative)) <= 1e-7 and grabbedinfo[i]._setBodyLinksToIgnore == testgrabbedinfo[i]._setBodyLinksToIgnore) for i in range(len(grabbedinfo))]):
```
Comment (cielavenir) [inline python/databases/linkstatistics.py]: this line also maybe

Comment (kanbouchou) [inline python/databases/linkstatistics.py]: thanks. should be fixed by 57e0b8b.

File: python/bindings/openravepy_kinbody.cpp (hunk 1 above)
Comment (kanbouchou) [inline python/bindings/openravepy_kinbody.cpp]: thanks. should be fixed by 57e0b8b.

**Comment by kanbouchou** [comment]
Also tested below
```
openrave.py -i original.dae
body1.Grab(body2, body1.GetLinks()[0])
env.Save('new.dae')
exit
openrave.py -i new.dae
body1.GetGrabbed()
```
and last line returns ``body2``, so saving and loading from dae also works.

I think this is ready for review now

**Comment by cielavenir** [comment]
Well, we should use kinbody_grab_gcc6 branch instead. There I merged gcc6 branch into the copy of kinbody_grab branch,

**Comment by rdiankov** [comment]
i think this is now good, thanks for the help guys!

**Pull request merged**